### PR TITLE
Add managed instances swagger to 2018-06-01-preview

### PR DESCRIPTION
**Pull request by johnpaulkee**
Add managed instances swagger to 2018-06-01-preview.
Main changes include moving ManagedInstances_ListByInstancePool over from 2015-05-01-preview over to 2018-06-01-preview, updating sql README.md to reference 2018-06-01-preview/managedInstances swagger.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 1]
```diff
@@ -484,6 +438,111 @@
...
+      "description": "The properties of managed instance maintenance window.",
+      "required": [
+        "frequency",
+        "customerTimeZone"
+      ],
+      "type": "object",
+      "properties": {
+        "timeOfUpgrade": {
+          "description": "Specifies time of upgrade for maintenance window of managed instance.",
+          "type": "string",
```
Comment (anuchandy) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: what is the format of this value? good to clarify that in the description. (e.g. 12-hour AM/PM, 24-hour clock)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/ManagedInstanceCreateMax.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        "subnetId": "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
+        "vCores": 8,
+        "storageSizeInGB": 1024,
+        "licenseType": "LicenseIncluded",
+        "collation": "SQL_Latin1_General_CP1_CI_AS",
+        "dnsZonePartner": "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Sql/managedInstances/testinstance",
+        "publicDataEndpointEnabled": false,
+        "proxyOverride": "Redirect",
+        "timezoneId": "UTC",
+        "instancePoolId": "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Sql/instancePools/pool1"
```
Comment (anuchandy) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/ManagedInstanceCreateMax.json]: is it possible to include example showing usage of `MaintenanceWindowSettings` property?

**Comment by anuchandy** [comment]
@jaredmoo SQL swagger please review.

Added "ARM review request" since the PR removes an API and introduces new property?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 3]
```diff
@@ -484,6 +438,111 @@
...
+          "$ref": "#/definitions/MaintenanceWindowSettings",
+          "description": "Specifies maintenance window settings for a managed instance.",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      }
+    },
+    "MaintenanceWindowSettings": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: Everything related to maintenance window isn't ready yet.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstances.json [hunk 4]
```diff
@@ -0,0 +1,745 @@
...
+          "$ref": "#/definitions/MaintenanceWindowSettings",
+          "description": "Specifies maintenance window settings for a managed instance.",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      }
+    },
+    "MaintenanceWindowSettings": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstances.json]: Also get rid of it here.

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Please address comments. Otherwise approved.

**Comment by johnpaulkee** [comment]
Will remove maintenance window changes from this PR.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/ManagedInstanceCreateMax.json (hunk 2 above)
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/examples/ManagedInstanceCreateMax.json]: Not needed as we're removing this property.

**Comment by johnpaulkee** [comment]
> Please address comments. Otherwise approved.

Thanks @jaredmoo - I've addressed comments.

**Comment by johnpaulkee** [comment]
> @jaredmoo SQL swagger please review.
> 
> Added "ARM review request" since the PR removes an API and introduces new property?

I removed the new property section.
The API is moved from 2015-05-01-preview to 2018-06-01-preview.

**Review by jaredmoo — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
@johnpaulkee - are you dropping support for this one API from 2015 preview? Whats the usage of this API in 2015 preview. For SQL, even though this API is marked as -preview, since it has been in this state for ever, its considered as perpetual preview. So, we don't want to break customers using this API -version. 
Are there other managedInstances APIs that will be moved to 2018 preview?

**Comment by johnpaulkee** [comment]
@ravbhatnagar currently there is no usage for this ManagedInstance_ListByInstancePool API in 2015-05-01-preview and we plan to support it in 2018-06-01-preview. Customer's won't be broken using the newer API version as there are no clients/customers using it yet. 

There are no other managed instance APIs that we are moving from 2015-05-01-preview, and support for 2018-06-01-preview already exists in production for all other managed instance APIs.

**Comment by ravbhatnagar** [comment]
Thanks for confirming @johnpaulkee . Signing off from ARM side since it was confirmed that there is no usage of this API and it wouldnt have worked since in ARM the supported API version is the new one.

**Comment by johnpaulkee** [comment]
> Thanks for confirming @johnpaulkee . Signing off from ARM side since it was confirmed that there is no usage of this API and it wouldnt have worked since in ARM the supported API version is the new one.

Thank you @ravbhatnagar.

**Comment by anuchandy** [comment]
Adding "breaking-change" tag since 2015-05-01-preview removes an API.

**Pull request merged**